addons/website_event/controllers/main.py

```diff
@@ -270,51 +264,59 @@ def _process_attendees_form(self, event, form_details):
             if int(ticket_id) not in event.event_ticket_ids.ids and len(event.event_ticket_ids.ids) > 0:
                 raise UserError(_("This ticket is not available for sale for this event"))
         registrations = {}
-        global_values = {}
         general_answer_ids = []
+        general_identification_answers = {}
+        already_treated_fields_data = {}
         for key, value in form_details.items():
-            if 'question_answer' in key and value:
-                dummy, registration_index, question_id = key.split('-')
-                question_sudo = request.env['event.question'].browse(int(question_id))
-                answer_values = None
-                if question_sudo.question_type == 'simple_choice':
-                    answer_values = {
-                        'question_id': int(question_id),
-                        'value_answer_id': int(value)
-                    }
-                elif question_sudo.question_type == 'text_box':
-                    answer_values = {
-                        'question_id': int(question_id),
-                        'value_text_box': value
-                    }
-
-                if answer_values and not int(registration_index):
-                    general_answer_ids.append((0, 0, answer_values))
-                elif answer_values:
-                    registrations.setdefault(registration_index, dict())\
-                        .setdefault('registration_answer_ids', list()).append((0, 0, answer_values))
-            else:
-                counter, attr_name = key.split('-', 1)
-                field_name = attr_name.split('-')[0]
+            if not value:
+                continue
+
+            key_values = key.split('-')
+            if len(key_values) == 2:
+                registration_index, field_name = key_values
+                registrations.setdefault(registration_index, dict())[field_name] = int(value)
+                continue
+
+            registration_index, question_type, question_id = key_values
+            answer_values = None
+            if question_type == 'simple_choice':
+                answer_values = {
+                    'question_id': int(question_id),
+                    'value_answer_id': int(value)
+                }
+            elif question_type != 'event_ticket_id':
```

Comment:
If I'm not mistaken we don't need this anymore, right? Could be a simple "else"?